tests/st/libnetwork/test_assign_specific_ip.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import uuid
+
+from tests.st.test_base import TestBase
+from tests.st.utils import utils
+from tests.st.utils.docker_host import DockerHost
+import logging
+from tests.st.utils.utils import assert_number_endpoints, assert_profile, \
+    get_profile_name, ETCD_CA, ETCD_CERT, ETCD_KEY, ETCD_HOSTNAME_SSL, \
+    ETCD_SCHEME
+from tests.st.libnetwork.test_mainline_single_host import \
+    ADDITIONAL_DOCKER_OPTIONS, POST_DOCKER_COMMANDS
+
+logger = logging.getLogger(__name__)
+
+
+class TestAssignIP(TestBase):
+    def test_assign_specific_ip(self):
+        """
+        Test that a libnetwork assigned IP is allocated to the container with
+        Calico when using the '--ip' flag on docker run.
+        """
+        with DockerHost('host',
+                        additional_docker_options=ADDITIONAL_DOCKER_OPTIONS,
+                        post_docker_commands=POST_DOCKER_COMMANDS,
+                        start_calico=False) as host:
+            host.start_calico_node("--libnetwork")
+
+            # Set up two endpoints on one host
+            workload1_ip = "192.168.1.101"
```

Comment:
I think this might be better as a 2-host test assigning IP addresses on each host in the same /23 network.  This will test the BGP CIDR exceptions and test that we can assign an address from a /23 that has already been claimed by a host... i.e. a little more bang for bucks.